### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by IhateTrains**
#92

### CODE REVIEW

**Comment by Idhrendur** [comment]
We need a CMakeLists (for the tests, I guess) first. There's nothing for CodeQL to build

**Comment by Idhrendur** [comment]
Looks to be a directories problem. CMakelists either needs to up a directory or the script needs to change directories first.

**Comment by Idhrendur** [comment]
Almost! Gotta switch to unix line endings.

**Comment by IhateTrains** [comment]
Should be working now.

File: tests/build_linux.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+# shellcheck disable=SC2103
+
+export CC=/usr/bin/gcc-10 &&
```
Comment (Zemurin) [inline tests/build_linux.sh:4]: Wasn't this supposed to be gcc-11?

Comment (IhateTrains) [reply tests/build_linux.sh:4]: In the branch there's currently nothing that requires GCC 11, but there will be if stringToInteger and stringToDouble are merged in. I will change it now.

**Review by Idhrendur — APPROVED** [review]

**Review by Zemurin — APPROVED** [review]

**Pull request merged**